File: arrays_and_strings/q3.py
import sys


if __name__ == "__main__":

    print("\nQ3: Given two strings, write a method to decide if one is a permutation of the other.\n\n")

    inputstrs = []

    if len(sys.argv) > 2:
        print("usage: python3 %s <string>" % sys.argv[0])
        sys.exit(-1)
    elif len(sys.argv) == 2:
        inputstrs.append(sys.argv[1])
    else:
        inputstrs = [("leolabs", "ballseo"), ("test", "tests"), ("dog", "god")]

    for input_tuple in inputstrs:
        assert len(input_tuple) == 2
        if sorted(input_tuple[0]) == sorted(input_tuple[1]):
            print("\nTRUE: %s is a permutation of %s\n" % (input_tuple[0], input_tuple[1]))
        else:
            print("\nFALSE: %s is NOT a permutation of %s\n" % (input_tuple[0], input_tuple[1]))

File: arrays_and_strings/q3.go
// Given two strings, write a method to decide if one is a permutation of the other.

package main

import (
	"github.com/thiagorobert/interviewprep/thiutil"
	"fmt"
	"reflect"
)

func main() {
	r := thiutil.NewReader()

	a := []byte(r.In())
	thiutil.Bsort(a)
	b := []byte(r.In())
	thiutil.Bsort(b)

	if reflect.DeepEqual(a, b) {
		fmt.Printf("%s and %s are anagrams", a, b)
	} else {
		fmt.Printf("%s and %s are garbage", a, b)
	}
}
